File: springboot-version/backend/src/main/java/com/petconnect/dto/PostDto.java
package com.petconnect.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostDto {
    private Long postId;
    private String title;
    private String description;
    private String imageUrl;
    private String visibility;
    private LocalDateTime createdAt;
    private Long userId;
    private String userFullName;
    private String userProfileImageUrl;
    private int likesCount;
    private List<CommentDto> comments;
}

File: aspnet-version/backend/PetConnectAlmost/DTOs/PostDto.cs
using PetConnectAlmost.Models;

namespace PetConnectAlmost.DTOs
{
    public class PostDto
    {
        public long PostId { get; set; }
        public long UserId { get; set; }
        public string UserFullName { get; set; } = string.Empty;  // From join
        [System.Text.Json.Serialization.JsonPropertyName("userProfileImageUrl")]
        public string? UserProfileImageUrl { get; set; } // From join
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public string Visibility { get; set; } = "PUBLIC";
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int LikesCount { get; set; }
        public List<CommentDto> Comments { get; set; } = [];
    }
}
